drizzle-kit/src/serializer/pgSerializer.ts

```diff
@@ -1934,7 +1935,37 @@ WHERE
 	};
 };
 
-const defaultForColumn = (column: any, internals: PgKitInternals, tableName: string) => {
+/**
+ * Formats a single array element based on the PostgreSQL column data type.
+ * Handles cleaning up quotes and spaces from postgres-array parsed elements,
+ * and applies appropriate formatting for different data types.
+ */
+const formatArrayElement = (element: any, dataType: string): string | null => {
+	if (element === null) return element;
+	
+	// Remove outer quotes from postgres-array parsed elements if present
+	// First trim spaces since postgres-array includes leading spaces in array elements
+	let cleanElement = typeof element === 'string' ? element.trim() : element;
+	if (typeof cleanElement === 'string' && ((cleanElement.startsWith("'") && cleanElement.endsWith("'")) || (cleanElement.startsWith('"') && cleanElement.endsWith('"')))) {
+		cleanElement = cleanElement.slice(1, -1);
+	}
+	
+	if (['integer', 'smallint', 'bigint', 'double precision', 'real'].includes(dataType.slice(0, -2))) {
```

Comment:
```suggestion
	if (['integer', 'smallint', 'bigint', 'double precision', 'real'].includes(dataType.slice(0, -"[]".length))) {
```

make the semantics a bit more clear (also would be nice to be safe and check if datatype ends with [])